### PR TITLE
VmWare: Don't repeat yourself with hardware devices

### PR DESCRIPTION
**Pull request by fwiesel**
The code was getting the hardware devices in multiple places directly,
and then (often, but not always) normalising the array.

Functions getting the device array then also had to normalise the array again.

By consolidating the retrieval and the normalisation in a function,
the code becomes less repetitive

### CODE REVIEW

File: nova/virt/vmwareapi/vm_util.py [hunk 1]
```diff
@@ -853,14 +853,22 @@ def _get_device_disk_type(device):
             return constants.DEFAULT_DISK_TYPE
 
 
+def _hardware_devices(hardware_devices):
```
Comment (joker-at-work) [inline nova/virt/vmwareapi/vm_util.py:856]: What do we need this split off for?

File: nova/virt/vmwareapi/vm_util.py [hunk 2]
```diff
@@ -853,14 +853,22 @@ def _get_device_disk_type(device):
...
+    if hardware_devices.__class__.__name__ == "ArrayOfVirtualDevice":
+        return hardware_devices.VirtualDevice
+    return hardware_devices
+
+
+def hardware_devices(session, vm_ref):
```
Comment (joker-at-work) [inline nova/virt/vmwareapi/vm_util.py:862]: I'd prefer `get_hardware_devices()` instead, but no must.

**Review by joker-at-work — COMMENTED** [review]
in the commit-message: `repetitive` (2 `e`, 2 `i`) from "repetition".

File: nova/virt/vmwareapi/vm_util.py (hunk 1 above)
Comment (fwiesel) [reply nova/virt/vmwareapi/vm_util.py:856]: It isn't really needed, it is the result of iterative refactoring. I first replaced all occurrences matching the code of `_hardware_devices`, and consolidated the `get_object_property(... "config.hardware.device")` calls.
After that, I removed all `_hardware_devices`, which got their output from `hardware_devices`.
And in the end only one call was left.

**Comment by fwiesel** [comment]
I've updated the commit-message

**Review by joker-at-work — APPROVED** [review]

**Review by joker-at-work — APPROVED** [review]

**Pull request merged**